src/js/utils/events.js

```diff
@@ -431,7 +431,7 @@ export function trigger(elem, event, hash) {
     trigger.call(null, parent, event, hash);
 
   // If at the top of the DOM, triggers the default action unless disabled.
-  } else if (!parent && !event.defaultPrevented) {
+  } else if (!parent && !event.defaultPrevented && event.target && event.target[event.type]) {
```

Comment:
this is the main fix?

Reply:
Looks like some tests are failing because of this. We probably want to expand the code above that makes sure we have an event object (line 413):

```js
  // If an event name was passed as a string, creates an event out of it
  if (typeof event === 'string') {
    event = {type: event, target: elem};
  } else if (!event.target) {
    event.target = elem;
  }
```

Reply:
Yeah this is the main fix

Reply:
It prevents `getData` from being called on something like `document`.